pallets/parachain-system/src/relay_state_snapshot.rs

```diff
@@ -164,20 +172,49 @@ impl RelayChainStateProof {
 	/// Read the [`MessagingStateSnapshot`] from the relay chain state proof.
 	///
 	/// Returns an error if anything failed at reading or decoding.
-	pub fn read_messaging_state_snapshot(&self) -> Result<MessagingStateSnapshot, Error> {
+	pub fn read_messaging_state_snapshot(
+		&self,
+		host_config: &AbridgedHostConfiguration,
+	) -> Result<MessagingStateSnapshot, Error> {
 		let dmq_mqc_head: relay_chain::Hash = read_entry(
 			&self.trie_backend,
 			&relay_chain::well_known_keys::dmq_mqc_head(self.para_id),
 			Some(Default::default()),
 		)
 		.map_err(Error::DmqMqcHead)?;
 
-		let relay_dispatch_queue_size: (u32, u32) = read_entry(
+		let relay_dispatch_queue_size = read_optional_entry::<RelayDispachQueueSize, _>(
 			&self.trie_backend,
-			&relay_chain::well_known_keys::relay_dispatch_queue_size(self.para_id),
-			Some((0, 0)),
-		)
-		.map_err(Error::RelayDispatchQueueSize)?;
+			&relay_chain::well_known_keys::relay_dispatch_queue_remaining_capacity(self.para_id)
+				.key,
+		);
+
+		// TODO paritytech/polkadot#6283: Remove all usages of `relay_dispatch_queue_size`
+		//
+		// When the relay chain and all parachains support `relay_dispatch_queue_remaining_capacity`,
+		// this code here needs to be removed and above needs to be changed to `read_entry` that
+		// returns an error if `relay_dispatch_queue_remaining_capacity` can not be found/decoded.
+		//
+		// For now we just fallback to the old dispatch queue size if there is an error.
+		let relay_dispatch_queue_size = match relay_dispatch_queue_size {
+			Ok(Some(r)) => r,
+			_ => {
+				let res = read_entry::<(u32, u32), _>(
```

Comment:
>It will require that all parachains migrate to relay_dispatch_queue_remaining_capacity before we can start removing the call to relay_dipatch_queue_size.

We cannot monitor this on the relay chain, or? Can we somehow know, how many parachains already support this?

Reply:
No, this needs to be done manually. Aka we wait one year at least.